### PR TITLE
Avoid confusing TS9025 error in `isolatedDeclarations`

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/60123

### CODE REVIEW

**Review by jakebailey — APPROVED** [review]
I think this is okay; erroring code will emit an error elsewhere, so letting `declaredParameterTypeContainsUndefined` "lie" a little to prevent follow-ons feels fine.

**Comment by jakebailey** [comment]
And in the case like:

```ts
// @ts-ignore
import { DoesNotExist } from "foo";

export class Bar {
    constructor(private x?: DoesNotExist | undefined) {}
}
```

The "worst thing" that happens here is that `| undefined` is spuriously left behind.

**Comment by jakebailey** [comment]
Ah, you already have that as a test via `Unresolved`.

File: tests/cases/compiler/isolatedDeclarationsAddUndefined2.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    constructor(private x: Array | undefined) {}
+}
+
+export class Bar4 {
+    constructor(private x: Array) {}
+}
+
+export function test1(x?: Array | undefined): void {}
+
+export function test2(x?: Unresolved | undefined): void {}
```
[the remark this answers is not in the document]
Comment (Andarist) [reply tests/cases/compiler/isolatedDeclarationsAddUndefined2.ts:25]: sure, i just pushed it out

**Pull request merged**